src/ars408_ros_node.cpp

```diff
@@ -85,18 +85,46 @@ radar_msgs::msg::RadarTrack PeContinentalArs408Node::ConvertRadarObjectToRadarTr
   return out_object;
 }
 
+radar_msgs::msg::RadarReturn PeContinentalArs408Node::ConvertRadarObjectToRadarReturn(
+  const ars408::RadarObject & in_object)
+{
+  radar_msgs::msg::RadarReturn radar_return;
+  radar_return.range = std::sqrt(
+    in_object.distance_long_x * in_object.distance_long_x +
+    in_object.distance_lat_y * in_object.distance_lat_y);
+  radar_return.azimuth = std::atan2(in_object.distance_lat_y, in_object.distance_long_x);
+  radar_return.doppler_velocity = in_object.speed_long_x / std::cos(radar_return.azimuth);
```

Comment:
Just a little confirmation.
I think atan2 returns value in [-π, π]. right?
Does this conversion match the definitions of azimuth and doppler_velocity in the RadarReturn Message?

Reply:
> I think atan2 returns value in [-π, π]. right?

Yes.

> Does the plus/negative sign from this conversion match the definitions of azimuth and doppler_velocity in the RadarReturn Message?

Yes.
In RadarScan.msg is based on sensor coordinate system.
If doppler_velocity is positive, the object move away from radar under any azimuth value.
And if doppler_velocity is negative, the object get closer to radar.

Just as your concerned, dependency to sensor coordinate doesn't make useful to use in Autoware.
So that, [radar_tracks_msgs_converter](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/radar_tracks_msgs_converter) handle transformation to baselink coordinate system.

Reply:
Thanks for the explanation!